File: 0775-Global and Local Inversions/0775-Global and Local Inversions.cpp
class Solution {
public:
    bool isIdealPermutation(vector<int>& A) {
        int n = A.size();
        int maxNum = A[0];
        for (int i = 0; i < n - 2; ++i) {
            maxNum = max(maxNum, A[i]);
            if (maxNum > A[i + 2]) return false;
        }

        return true;
    }
};

File: 0701-0800/0775-Global and Local Inversions/0775-Global and Local Inversions.py
class Solution:
    def isIdealPermutation(self, A: List[int]) -> bool:
        maxNum = A[0]
        for i in range(len(A) - 2):
            maxNum = max(maxNum, A[i])
            if maxNum > A[i + 2]:
                return False
        return True

class Solution2:
    def isIdealPermutation(self, A: List[int]) -> bool:
        return all(abs(a - i) <= 1 for i, a in enumerate(A))
